src/app/social-media/social-media-list-input/social-media-list-input.component.ts

```diff
@@ -0,0 +1,167 @@
+          formArray.push(this.createForms(item));
+          console.log(item);
+        });
+        this.socialMediaPostListForm.setControl('postList', formArray);
+      }
+
+    }
+  }
+
+  get socialMediaPosts(): FormArray {
+    console.log();
+    return this.socialMediaPostListForm.get('postList') as FormArray;
+  }
+
+  addSocialMediaPost() {
+    this.socialMediaPosts.push(new FormGroup(
+      {
+        id: new FormControl(),
+        url: new FormControl('', [Validators.required, Validators.pattern(reg)]),
+        order: new FormControl()
+      }
+    ));
+  }
+
+  deleteSocialMediaPost(index: number) {
+    let post = this.socialMediaPosts.at(index).value;
+
+    const modal = this.modal.open(DeletePostConfirmationModalComponent, { size: 'lg' });
+    modal.componentInstance.url = post.url;
+
+    modal.result.then((result) => {
+      console.log(result);
+      if (result == 'Confirm') {
+        this.socialMediaPosts.removeAt(index);
+        if ((post.id !== 'undefined') && (post.id !== null)) {
+          this.socialMediaService.deleteSocialMediaPost(post.id).subscribe(
+            () => {
+              console.log('delete post id ' + post.id + ' success');
+            },
+            () => {
+              console.log("Failed to delete post " + post.id);
+            }
+          );
+        }
+      }
+    }, (reason) => {
+    });
+
+  }
+
+  move(shift, currentIndex) {
+
+    let newIndex: number = currentIndex + shift;
+    if (newIndex === -1) {
+      newIndex = this.socialMediaPosts.length - 1;
+    } else if (newIndex == this.socialMediaPosts.length) {
+      newIndex = 0;
+    }
+
+    const currentGroup = this.socialMediaPosts.at(currentIndex);
+    this.socialMediaPosts.removeAt(currentIndex);;
+    this.socialMediaPosts.insert(newIndex, currentGroup)
+  }
+
+  submit() {
+    console.log(this.socialMediaPosts);
+    if (this.socialMediaPosts.valid) {
+      this.socialMediaPosts.value.forEach((post, index) => {
+        post = post as SocialMediaPost;
+        post.sortOrder = index;
+        if (post.id != null) {
+
+          this.socialMediaService.updateSocialMediaPost(post.id, post).subscribe(
+            () => {
+              console.log('update success: ' + post.id);
+            },
+            () => {
+              console.log("Failed to update psot: " + post.id);
+            }
+          );
+        }
+        else {
+          delete post["id"];
+          this.socialMediaService.addSocialMediaPost(post).subscribe(
+            () => {
+              console.log('create success');
+            },
+            () => {
+              console.log("Failed to create post");
+            }
+          );
+        }
+      });
+    }
+    else{
+      /// TODO: if submit failed, do something
+      console.log('not valid');
+    }
+  }
+
+  get_url(index: number): any{
+    let arrayControl = this.socialMediaPostListForm.get('postList') as FormArray;
+    let item = arrayControl.at(index);
```

Comment:
not sure about the name of the variable. How about directly returning the result?